components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/RegistryPolicyDAOImpl.java

```diff
@@ -342,44 +360,18 @@ public String getReferencedPolicy(String policyId) {
     @Override
     public Map<String, Set<AttributeDTO>> getSearchAttributes(String identifier, Set<AttributeDTO> givenAttribute) {
 
-        PolicyDTO[] policyDTOs = null;
-        Map<String, Set<AttributeDTO>> attributeMap = null;
+        PolicyDTO[] policyDTOs;
         try {
             policyDTOs = getAllPolicies(true, true);
-        } catch (Exception e) {
-            LOG.error("Policies can not be retrieved from registry policy finder module", e);
-        }
 
-        if (policyDTOs != null) {
-            attributeMap = new HashMap<>();
-            for (PolicyDTO policyDTO : policyDTOs) {
-                Set<AttributeDTO> attributeDTOs =
-                        new HashSet<>(Arrays.asList(policyDTO.getAttributeDTOs()));
-                String[] policyIdRef = policyDTO.getPolicyIdReferences();
-                String[] policySetIdRef = policyDTO.getPolicySetIdReferences();
-
-                if (policyIdRef != null && policyIdRef.length > 0 || policySetIdRef != null &&
-                        policySetIdRef.length > 0) {
-                    for (PolicyDTO dto : policyDTOs) {
-                        if (policyIdRef != null) {
-                            for (String policyId : policyIdRef) {
-                                if (dto.getPolicyId().equals(policyId)) {
-                                    attributeDTOs.addAll(Arrays.asList(dto.getAttributeDTOs()));
-                                }
-                            }
-                        }
-                        for (String policySetId : policySetIdRef) {
-                            if (dto.getPolicyId().equals(policySetId)) {
-                                attributeDTOs.addAll(Arrays.asList(dto.getAttributeDTOs()));
-                            }
-                        }
-                    }
-                }
-                attributeMap.put(policyDTO.getPolicyId(), attributeDTOs);
+            if (policyDTOs.length > 0) {
```

Comment:
Can `policyDTOs` be null? Coz there is a null check in the previous code.

Reply:
Method `getAllPolicies()` will not return null, so this cannot be null.